sql/src/test/java/org/apache/druid/sql/calcite/CalciteIngestionDmlTest.java

```diff
@@ -299,13 +363,23 @@ private void verifySuccess()
           .expectedResources(expectedResources)
           .run();
 
+      String expectedLogicalPlan;
+      if (expectedLogicalPlanResource != null) {
+        expectedLogicalPlan = StringUtils.getResource(
+            this,
+            "/calcite/expected/ingest/" + expectedLogicalPlanResource + "-logicalPlan.txt"
```

Comment:
is this a bit fragile since it requires naming files correctly? (i don't really have any better ideas, just thinking out loud)

Reply:
Yes, that is true. If we had Java 17, we could easily put the text inline in the code. I thought about fiddling with quotes, or building up a chunk of text from a list of lines, but in the end said, the heck with it, let's just use resource files.

In code that uses resource files with tests, it is common to define a directory structure to organize the files. This like creates the "structure" for the Calcite DML tests: the stuff goes in that one folder.

This doesn't generalize (yet): if any other Calcite test (other than DML) wants to use resources, then there is no defined place to put the files. We could move this higher in the class hierarchy. (Or, wait for @imply-cheddar to commit his file-based test solution.)